service/matching/user_data_manager.go

```diff
@@ -322,11 +323,20 @@ func (m *userDataManagerImpl) fetchUserData(ctx context.Context) error {
 		// spinning. So enforce a minimum wait time that increases as long as we keep getting
 		// very fast replies.
 		if elapsed < m.config.GetUserDataMinWaitTime {
-			util.InterruptibleSleep(ctx, minWaitTime-elapsed)
-			// Don't let this get near our call timeout, otherwise we can't tell the difference
-			// between a fast reply and a timeout.
-			minWaitTime = min(minWaitTime*2, m.config.GetUserDataLongPollTimeout()/2)
+			if fastResponseCounter >= 3 {
```

Comment:
I like this, though now I want to refactor the code to make it cleaner. later though